src/lib/testing/expect-bound-function.ts

```diff
@@ -7,15 +7,25 @@ import { createTestEnvironment } from './create-test-env';
 import { Mocks } from './mocks';
 
 interface WithReducerStage<State, Args extends any[], Return> {
-  withReducer(reducer: Reducer<State, any>, initialState?: DeepPartial<State>): WithMocksStage<State, Args, Return>;
+  /**
+   * Sets the `reducer` for the test environment
+   *
+   * @param reducer
+   * @param initialState Initial state. If none is provided,
+   * the `reducer` will be called once without an action to generate the initial state
+   */
+  withReducer(
+    reducer: Reducer<State, any>,
+    initialState?: DeepPartial<State>,
+  ): CallStage<State, Args, Return> & WithMocksStage<State, Args, Return>;
```

Comment:
Why do we need the `CallStage<State, Args, Return> &` here? The `extends` in the `WithMocks` stage should take care of it

Reply:
Right, didn't expect that `extends` there.

I think it'd rather remove the `extends` though and keep this here, since that'd just affect a single case and not be the default (that mock could always be skipped).

Reply:
Ah, but the same pattern is used in the `AssertionState`. Then I'll leave it as is for now and change the return value.

Reply:
Sorry, botched the merge in between, but then cherry-picked the docs changes again. Didn't know they were happening in here.

Resorted to non-inheriting interfaces, and only have the return type be unions. The test implementation didn't need updated annotations, as all interfaces are now explicitly implement on it, and it always returns `this`.

Reply:
@HenriBeck Ok for you? From my side we could merge this.